evalml/model_understanding/graphs.py

```diff
@@ -890,3 +891,73 @@ def graph_prediction_vs_actual_over_time(pipeline, X, y, dates):
                         yaxis={'title': 'Target Values and Predictions'})
 
     return _go.Figure(data=data, layout=layout)
+
+
+def t_sne(X, n_components=2, perplexity=30.0, learning_rate=200.0, metric='euclidean', **kwargs):
+    """Get the transformed output after fitting X to the embedded space.
+
+     Arguments:
+        X (np.ndarray, ww.DataTable, pd.DataFrame): Data to be transformed.
+        n_components (int, optional): Dimension of the embedded space.
+        perplexity (float, optional): Related to the number of nearest neighbors that is used in other manifold learning
+        algorithms. Larger datasets usually require a larger perplexity. Consider selecting a value between 5 and 50.
+        learning_rate (float, optional): Usually in the range [10.0, 1000.0]. If the cost function gets stuck in a bad
+        local minimum, increasing the learning rate may help.
+        metric (str, optional): The metric to use when calculating distance between instances in a feature array.
+
+    Returns:
+        np.ndarray (n_samples, n_components)
+    """
+    if not isinstance(n_components, int) or not n_components > 0:
+        raise ValueError("The parameter n_components must be of type integer and greater than 0")
+    if not perplexity >= 0:
+        raise ValueError("The parameter perplexity must be non-negative")
+
+    X = _convert_to_woodwork_structure(X)
+    X = _convert_woodwork_types_wrapper(X.to_dataframe())
+    t_sne_ = TSNE(n_components=n_components, perplexity=perplexity, learning_rate=learning_rate, metric=metric, **kwargs)
+    X_new = t_sne_.fit_transform(X)
+    return X_new
+
+
+def graph_t_sne(X, n_components=2, perplexity=30.0, learning_rate=200.0, metric='euclidean', marker_line_width=2, marker_size=7, **kwargs):
+    """Visualization tool for high dimensional data.
```

Comment:
nit-pick: I think we should make this for specific like, "Plot projection of data onto lower dimensional space with TSNE algorithm" or something like that?